src/sinks/azure_common/config.rs

```diff
@@ -121,34 +121,44 @@ pub fn build_client(
     connection_string: Option<String>,
     storage_account: Option<String>,
     container_name: String,
+    blob_endpoint: Option<String>,
 ) -> crate::Result<Arc<ContainerClient>> {
     let client;
     match (connection_string, storage_account) {
         (Some(connection_string_p), None) => {
             let connection_string = ConnectionString::new(&connection_string_p)?;
-            let mut builder = ClientBuilder::new(
-                connection_string
-                    .account_name
-                    .ok_or("Account name missing in connection string")?,
-                connection_string.storage_credentials()?,
-            )
-            .retry(RetryOptions::none());
-            if let Some(uri) = connection_string.blob_endpoint {
-                builder = builder.cloud_location(CloudLocation::Custom {
+
+            let builder = match connection_string.blob_endpoint {
+                Some(uri) => ClientBuilder::with_location(CloudLocation::Custom {
                     uri: uri.to_string(),
                     credentials: connection_string.storage_credentials()?,
-                });
+                }),
+                None => ClientBuilder::new(
+                    connection_string
+                        .account_name
+                        .ok_or("Account name missing in connection string")?,
+                    connection_string.storage_credentials()?,
+                ),
             }
+            .retry(RetryOptions::none());
+
             client = builder.container_client(container_name);
         }
         (None, Some(storage_account_p)) => {
             let creds = std::sync::Arc::new(DefaultAzureCredential::default());
             let auto_creds = std::sync::Arc::new(AutoRefreshingTokenCredential::new(creds));
             let storage_credentials = StorageCredentials::TokenCredential(auto_creds);
 
-            client = ClientBuilder::new(storage_account_p, storage_credentials)
-                .retry(RetryOptions::none())
-                .container_client(container_name)
+            let builder = match blob_endpoint {
+                Some(blob_endpoint) => ClientBuilder::with_location(CloudLocation::Custom {
```

Comment:
Technically `storage_account_p` is not used when a `blob_endpoint` is, due to the way that the `CloudLocation` works, but I didn't want to add additional logic for managing a third case and duplicate the code necessary here for the storage_credentials setup.

I can if necessary though and make `blob_endpoint` mutually exclusive with `storage_account` and `connection_string`.